src/main/scala/za/co/absa/spark/commons/implicits/ArrayTypeImplicits.scala

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2021 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.spark.commons.implicits
+
+import org.apache.spark.sql.types.{ArrayType, DataType, StructType}
+import za.co.absa.spark.commons.implicits.StructTypeImplicits.StructTypeEnhancements
+
+import scala.annotation.tailrec
+
+object ArrayTypeImplicits {
+
+  implicit class ArrayTypeEnhancements(arrayType: ArrayType) {
+
+    /**
+     * Compares 2 array fields of a dataframe utils.
+     *
+     * @param other The second array to compare
+     * @return true if provided arrays are the same ignoring nullability
+     */
+    @scala.annotation.tailrec
+    final def isEquivalentArrayType(other: ArrayType): Boolean = {
```

Comment:
This seem *very* similar to `DataTypeEnhancements(dt: DataType).isEquivalentDataType()`. Couldn't one of these methods use the other instead of containing the same logic?

Reply:
Fine by me, makes sense.